### PR TITLE
RDM-2438: Fix classifications on simple collections

### PR DESCRIPTION
**Pull request by vlaurin**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-2438

### Change description ###

Fixes RDM-2438

Classifications for simple collections were store as `classification` but read as `value`. As such, classifications were not found and by default items were filtered out.
Add fallback to read classification from `classification` node

This is a quick fix to restore the use of simple collections. Additional work around classifications on collections will be provided as part of RDM-882.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by vlaurin** [comment]
I've now fixed the issue with the classifications on simple collection items being set to `null` during updates. However, this comes with a significant increase in changed lines, @andrewfolga could I trouble you to re-review please?

File: src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseDataService.java [hunk 1]
```diff
@@ -110,17 +110,24 @@ private void deduceClassificationForCollectionType(JsonNode dataNode, JsonNode e
...
-                        getExistingDataClassificationFromArrayOrEmpty(existingDataClassificationNode, field),
+                        itemClassification,
                         caseField.getFieldType().getCollectionFieldType().getComplexFields(),
                         fieldIdPrefix + fieldName + FIELD_SEPARATOR);
                 } else {
                     final ObjectNode simpleCollectionItemNode = (ObjectNode) field;
                     // Add `classification` property
-                    deduceClassificationForSimpleType(simpleCollectionItemNode, existingDataClassificationNode, CLASSIFICATION, caseField);
+                    final String newClassification = itemClassification.isTextual() ?
+                        itemClassification.textValue() : getClassificationFromCaseFieldOrEmpty(caseField);
```
Comment (andrewfolga) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseDataService.java]: Not related to your changes but can we change the name of the getClassificationFromCaseFieldOrEmpty to something closer to truth like getClassificationFromCaseFieldOrDefault?

File: src/test/java/uk/gov/hmcts/ccd/domain/service/common/CaseDataServiceTest.java [hunk 2]
```diff
@@ -4,73 +4,78 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
 import uk.gov.hmcts.ccd.data.casedetails.SecurityClassification;
 import uk.gov.hmcts.ccd.domain.model.definition.CaseField;
 import uk.gov.hmcts.ccd.domain.model.definition.CaseType;
+import uk.gov.hmcts.ccd.domain.model.definition.FieldType;
 
 import java.io.IOException;
+import java.util.Arrays;
```
Comment (andrewfolga) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/common/CaseDataServiceTest.java]: Unused import

File: src/test/java/uk/gov/hmcts/ccd/domain/service/common/CaseDataServiceTest.java [hunk 3]
```diff
@@ -4,73 +4,78 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.collect.Maps;
+import org.hamcrest.MatcherAssert;
```
Comment (andrewfolga) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/common/CaseDataServiceTest.java]: unused import

File: src/test/java/uk/gov/hmcts/ccd/domain/service/common/CaseDataServiceTest.java [hunk 4]
```diff
@@ -4,73 +4,78 @@
...
+import org.hamcrest.Matchers;
```
Comment (andrewfolga) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/common/CaseDataServiceTest.java]: unused import

**Comment by Saravanabh** [comment]
This PR has been tested in local and works fine. @ngul can you merge this please.

**Comment by Saravanabh** [comment]
This has been tested on my local env and works as expected. Ready to be merged.

**Pull request merged**